oprf-client/src/lib.rs

```diff
@@ -114,554 +48,167 @@ pub enum Error {
-        signal_hash,
-        id_commitment_r,
-        rng,
-    )
-}
+    let share_identifier = ShareIdentifier {
+        oprf_key_id,
+        share_epoch,
+    };
 
-/// Signs an OPRF query and prepares it for sending to OPRF peers.
-///
-/// This function performs the following steps:
-/// 1. Validates inputs, including Merkle tree depth and public key index.
-/// 2. Generates a query hash from the Merkle index, RP identifier, and requested action.
-/// 3. Blinds the query hash using the user's public key and a random blinding factor.
-/// 4. Signs the blinded query with the user’s private key.
-/// 5. Constructs a [`QueryProofInput`] containing the signature, credential data,
-///    Merkle membership information, and other metadata required for the proof.
-/// 6. Generates the OPRFQuery zero-knowledge proof using the provided
-///    proving key material.
-///
-/// # Arguments
-///
-/// * `credentials_signature` - The user's credential signature issued by the RP.
-/// * `merkle_membership` - Used to compute proof for membership in the Merkle tree.
-/// * `query_material` - Groth16 proving key and constraint matrices for query proof.
-/// * `query` - The query details (RP ID, action, nonce, timestamp).
-/// * `key_material` - User key material including private signing key.
-/// * `rng` - Cryptographically secure random number generator.
-///
-/// # Errors
-///
-/// Returns an [`Error`] if:
-/// - The public key index is out of bounds.
-/// - Groth16 proof generation fails.
-///
-/// # Returns
-///
-/// A [`SignedOprfQuery`] containing:
-/// - The generated `OprfRequest` ready to initiate sessions with OPRF peers.
-/// - The blinding factor and query hash used for later computations.
-/// - The Groth16 proof input for verification in the nullifier step.
-pub fn sign_oprf_query<R: Rng + CryptoRng>(
-    credentials_signature: CredentialsSignature,
-    merkle_membership: MerkleMembership,
-    query_material: &CircomGroth16Material,
-    query: OprfQuery,
-    key_material: UserKeyMaterial,
-    request_id: Uuid,
-    rng: &mut R,
-) -> Result<SignedOprfQuery> {
-    if key_material.pk_index >= MAX_PUBLIC_KEYS as u64 {
-        return Err(Error::InvalidPublicKeyIndex(key_material.pk_index));
-    }
+    let (blinded_request, blinding_factor) =
+        oprf_core::oprf::client::blind_query(query, blinding_factor);
+    let oprf_req = OprfRequest {
+        request_id,
+        blinded_query: blinded_request.blinded_query(),
+        share_identifier,
+        auth,
+    };
 
-    let query_hash = oprf::client::generate_query(
-        merkle_membership.mt_index.into(),
-        query.rp_id.into_inner().into(),
-        query.action,
-    );
-    let (blinded_request, blinding_factor) = oprf::client::blind_query(query_hash, rng);
-    let signature = key_material.sk.sign(blinding_factor.query());
+    tracing::debug!("initializing sessions at {} services", services.len());
+    let reqwest_client = reqwest::Client::new();
```

Comment:
This should be improved later on as well, we do not want to build this client here all the time.
I would suggest moving this back to a struct with some members and methods instead of freestanding functions TBH. We can also do it in a separate PR though